### ISSUE TITLE
Too many images being reported by Google from 2.4.1.1

### ISSUE THREAD

**Issue by michaeltorbert** [semperfiwebdesign/all-in-one-seo-pack#1252]
Reported by bobsled here: https://wordpress.org/support/topic/a-lot-of-images-with-version-2-4-1-1/

> I have updated to Version 2.4.1.1 with no problems or errors. However, the image count shown on Webmaster Tools is very high. My WordPress media library counts 1,625 images, but Google reports 5,459. Is this expected behaviour?

**Comment by contactashish13** [comment]
@michaeltorbert this could be expected behavior if the user wants to show all post types, including attachment types. So if he has uploaded X images and attached one each to a post thus making X posts, the sitemap will show 2X images - one for the posts and one for the attachment type.

**Comment by michaeltorbert** [comment]
@wpsmort Can you compare apples to apples with other plugins and see if they have the same results in Google Search Console?

**Comment by wpsmort** [comment]
@contactashish13 i tested PR #1254 and it does reduce the number of images down significantly, however, there is still a problem where an image is added to the sitemap twice.  This happens when the image tag looks like this:

```
<img class="alignleft wp-image-18608 size-full" src="http://dahaines.com/wp-content/uploads/2016/08/Blue-Days-Small.jpg" alt="Blue Days by Derek Haines" width="375" height="375" srcset="http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small.jpg 375w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-300x300.jpg 300w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-270x270.jpg 270w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-230x230.jpg 230w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-150x150.jpg 150w" sizes="(max-width: 375px) 100vw, 375px" />
```
When that tag is found, we add the following to the sitemap:
```
<image:image>
	<image:loc>http://derekhaines-clone.dev/wp-content/uploads/2016/08/Blue-Days-Small.jpg</image:loc>
</image:image>
<image:image>
	<image:loc>http://dahaines.com/wp-content/uploads/2016/08/Blue-Days-Small.jpg</image:loc>
</image:image>
```

Notice how one image:loc tag has the URL of my local test site and the other has the URL for the customer's actual live site.

**Comment by wpsmort** [comment]
Notes from testing:  The code in PR #1254 reduced the number of images shown in the sitemap by this amount:

v2.4.1.1 code:
sitemap_post.xml = 3198 images
sitemap_page.xml = 54 images

PR 1254 code:
sitemap_post.xml = 2051 images
sitemap_page.xml = 30 images

**Comment by wpsmort** [comment]
@michaeltorbert I tested with a different sitemap plugin and these are the number of images it found:

post-sitemap.xml = 1629 images
page-sitemap.xml = 27 images
attachment-sitemap.xml = 1617 images

The difference was because we're adding some images twice, see my comment here - https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1252#issuecomment-333893174

**Comment by contactashish13** [comment]
@wpsmort I tested with the below HTML snippet and it adds only one image. I don't think this could be the issue because if we happened to take the images mentioned in the "srcset" attribute, you would find 5 more images. I believe the issue is somewhere else.

```
<img class="alignleft wp-image-18608 size-full" src="http://dahaines.com/wp-content/uploads/2016/08/Blue-Days-Small.jpg" alt="Blue Days by Derek Haines" width="375" height="375" srcset="http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small.jpg 375w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-300x300.jpg 300w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-270x270.jpg 270w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-230x230.jpg 230w, http://derekhaines.dev/wp-content/uploads/2016/08/Blue-Days-Small-150x150.jpg 150w" sizes="(max-width: 375px) 100vw, 375px" />
```

**Comment by michaeltorbert** [comment]
We'll go ahead and include PR #1254 in 2.4.2 since it at least improves the image count so we aren't holding up the release, then we'll revisit this for 2.4.3.

**Comment by michaeltorbert** [comment]
@contactashish13 Is there a PR for this?

**Comment by contactashish13** [comment]
PR https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/1254

**Comment by wpsmort** [comment]
@contactashish13 That PR was merged when releasing v2.4.2.  This needs a new PR for any new changes you've made.

**Comment by contactashish13** [comment]
@wpsmort I'm not sure how to check this. I see that all my changes were in the PR and I did not add anything after it was merged.

**Comment by wpsmort** [comment]
I've tested again with various ways of embedding images and the image count in Google Webmaster Tools matches the image count in my XML Sitemap which matches the number of images on my site.  I'd say this was fixed in v2.4.2 and we can close this issue.  We've had no new reports of this problem.
